sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -315,8 +317,14 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
         assert(
           joinOperator.right.find(_.isInstanceOf[SortExec]).isDefined == sortRight,
           s"expected sort in the right child to be $sortRight but found\n${joinOperator.right}")
+
+        if (expectedResult.isDefined) {
+          checkAnswer(joined, expectedResult.get)
```

Comment:
In L296 we already have an `assert` to check the result, do we really need this?